cli/azd/pkg/project/framework_service_docker.go

```diff
@@ -619,3 +630,29 @@ func getDockerOptionsWithDefaults(options DockerProjectOptions) DockerProjectOpt
 
 	return options
 }
+
+// todo: hardcode jdk-21 as base image here, may need more accurate java version detection.
+const DefaultDockerfileForJavaProject = `FROM openjdk:21-jdk-slim
+COPY ./target/*.jar app.jar
+ENTRYPOINT ["sh", "-c", \
+    "if [ -f /app.jar ]; then java -jar /app.jar; \
+    else echo 'Error: No JAR file found' >&2; exit 1; fi"]`
```

Comment:
Is it possible to report this error at build time instead of runtime?

Reply:
It's in the bash script, normally it returns non-zero code to indicate any error. I don't have any other better idea for now. 

BTW, I think we can have the assumption that after successfully `mvn clean package` (we added the hook), the output jar should be located at the `target` folder, so even we don't need the check in the line 639, just `java -jar *.jar`.

Reply:
> so even we don't need the check in the line 639, just java -jar *.jar is enough?

Agree, we can change to this:
```shell
ENTRYPOINT ["sh", "-c", "java -jar /app.jar"]
```

BTW, please confirm after using base image `openjdk:21-jdk-slim`, the working directory is root dir `/`. So after `COPY ./target/*.jar app.jar`, the `app.jar` will be put in root dir. I suggest to change `COPY ./target/*.jar app.jar` to `COPY ./target/*.jar /app.jar` to make our code more robust.

Reply:
> It's in the bash script, normally it returns non-zero code to indicate any error. I don't have any other better idea for now. What do you think about it?

If `./target/*.jar` not exist, will `COPY ./target/*.jar app.jar` make `docker build` failed?

Reply:
> change `COPY ./target/*.jar app.jar` to `COPY ./target/*.jar /app.jar` to make our code more robust.

The default working dir is `/`, and also updated the COPY command.

> If `./target/*.jar` not exist, will `COPY ./target/*.jar app.jar` make docker build failed?

Test from my side, it will throw an error:
![image](https://github.com/user-attachments/assets/85923527-49e3-47ec-98df-82c7bdada9d3)